bin/remote-prover/src/generated/client/mod.rs

```diff
@@ -0,0 +1,26 @@
+//! Client-side generated protobuf code module
+//!
+//! This module provides conditional access to client protobuf definitions
+//! based on the std/no_std environment.
+
+#[cfg(feature = "std")]
+pub mod std;
+
+#[cfg(not(feature = "std"))]
+pub mod nostd;
+
+// Re-export based on current environment
+#[cfg(feature = "std")]
+pub use std::*;
+
+#[cfg(not(feature = "std"))]
+pub use nostd::*;
```

Comment:
I don't think we should have both `pub mod x` and `pub use x::*`.

Definitely `pub mod std` is a no-no because that will trivially clash with the Rust `std::` paths.